core/test/oracle/Voting.js

```diff
@@ -170,6 +174,15 @@ contract("Voting", function(accounts) {
     // Move to the reveal phase.
     await moveToNextPhase(voting);
 
+    // Can't snapshot unless you prove you are EOA with signature
+    assert(await didContractThrow(voting.snapshotCurrentRound()));
+
+    // Can't reveal if snapshot has not been taken yet, even if all data is correct
+    assert(await didContractThrow(voting.revealVote(identifier, time, newPrice, newSalt)));
+
+    // This is now required before reveal
+    await voting.snapshotCurrentRound(signature);
+
```

Comment:
Unrelated to this line -- just wanted to put this comment somewhere.

I would suggest a new test `it` that just tests different signatures:
- Valid signature should not revert.
- Signature that's just random bytes should revert ([web3.utils.randomHex](https://web3js.readthedocs.io/en/v1.2.6/web3-utils.html#randomhex) will allow you to generate random bytes).
- Signature from a different account should revert.
- Signature with incorrect message should revert (this one's less important bc this isn't really an attack vector).

Note: the setup for this test should be very easy -- you don't need all the prep from the other tests. All you should need to do is push the contract to the reveal phase (can be done by calling moveToNextRound followed by moveToNextPhase) and call `snapshotCurrentRound()`. No price requests or commits should be necessary.